crates/rolldown_ecmascript/src/ecma_ast/mod.rs

```diff
@@ -38,6 +39,25 @@ impl EcmaAst {
   pub fn comments(&self) -> &oxc::allocator::Vec<'_, Comment> {
     &self.program.borrow_dependent().program.comments
   }
+
+  /// Clone the `Program` with another `Allocator`.
+  /// and copy rest fields, this is used to cache the Ast in incremental compilation
+  /// use another `Allocator` to avoid memory leak because.
+  #[must_use]
+  pub fn clone_with_another_arena(&self) -> EcmaAst {
```

Comment:
Since `EcmaAst` is consider a owned type, How about just use `impl Clone`?

Reply:
This is intended to avoid using `Clone` trait impl, because people may accidentally misuse the `clone` method, and meet some unexpected memory issue.